client/landing/stepper/declarative-flow/flows/site-setup-flow/site-setup-flow.ts

```diff
@@ -88,6 +89,35 @@ const siteSetupFlow: FlowV1 = {
 
 		return steps;
 	},
+	useAlternateStep( { currentStepRoute, steps } ) {
+		const urlQueryParams = useQuery();
+		const intent = urlQueryParams.get( 'intent' );
+		const importPlatform = urlQueryParams.get( 'importPlatform' );
```

Comment:
I think here we can have a very good use of pattern matches

https://www.telerik.com/blogs/pattern-matching-typescript 

match(queryParams)
  .with({ intent: "import", platform: P.string() }, ({platform}) =>   getPlatformName(platform) )
   .with({ intent: "import",  () =>  `importList" )
   .otherwise(undefined)
   .exhaustive();

Reply:
💯  I definitely think this area could be improved upon to make it easier to understand and maintain -- this was definitely a first pass to show what logic could be used/useful in practice.